external/fv3fit/fv3fit/pytorch/training_loop.py

```diff
@@ -0,0 +1,76 @@
+import dataclasses
+import logging
+import numpy as np
+import torch
+import tensorflow_datasets as tfds
+
+logger = logging.getLogger(__name__)
+
+
+def evaluate_model(config, model, data_iter):
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+    loss = config.loss()  # torch.nn.MSELoss()
+    model.eval()
+    l_sum, n = 0.0, 0
+    with torch.no_grad():
+        for x, y in data_iter:
+            y_pred = model(torch.as_tensor(np.squeeze(x)).float().to(device))
+            y = torch.as_tensor(np.squeeze(y)).float().to(device)
+            ll = loss(y_pred, y)
+            l_sum += ll.item() * y.shape[0]
+            n += y.shape[0]
+        return l_sum / n
+
+
+@dataclasses.dataclass
+class TrainingLoopConfig:
+    """
+    Attributes:
+        epochs: number of times to run through the batches when training
+        shuffle_buffer_size: size of buffer to use when shuffling samples
+        save_path: name of the file to save the best weights
```

Comment:
You can keep this for now until we have dump and load implemented, but saving the trained model is supposed to be the responsibility of the top level, where we call `predictor.dump`. This should be removed when we've added a `dump` and `load` method to your predictor

Reply:
I think this is trying to replicate the keras callback we use that checkpoints the model.  Might need some kind of model wrapper to emulate the checkpointing without including in the responsibilities of the training loop